src/functions.js

```diff
@@ -516,15 +517,42 @@ defineFunction([
     };
 });
 
-// Left and right overlap functions
-defineFunction(["\\llap", "\\rlap"], {
+// Horizontal overlap functions
+defineFunction(["\\mathllap", "\\mathrlap", "\\mathclap"], {
     numArgs: 1,
     allowedInText: true,
 }, function(context, args) {
     const body = args[0];
     return {
-        type: context.funcName.slice(1),
+        type: "lap",
+        className: context.funcName.slice(5),
+        body: body,
+    };
+});
+
+// smash, with optional [bt], as in AMS
+defineFunction("\\smash", {
+    numArgs: 1,
+    numOptionalArgs: 1,
+    allowedInText: true,
+}, function(context, args) {
+    const tbArg = args[0];
+    let tb = "";
+    if (tbArg) {
+        let letter = "";
+        for (let i = 0; i < tbArg.value.length; ++i) {
+            letter = tbArg.value[i].value;
+            if (letter === "t" || letter === "b") {
+                tb += letter;
+            }
+        }
+    }
+
+    const body = args[1];
+    return {
+        type: "smash",
         body: body,
+        tb: tb,
```

Comment:
```
smashHeight: /t/.test(tbArg),
smashDepth: /b/.test(tbArg),
```

Reply:
Do we care if there are other values in `tbArg`?

Reply:
Yes, I suppose we do care. I just checked and AMS LaTeX ignores the `\smash` if I put another character in there. Here's an example and the first exponent has `\smash[tab]{2}`
![](http://quicklatex.com/cache3/5b/ql_147b16a8b69c78a5665bb5a89345fe5b_l3.svg)

I'll change the code to act in the same way.